cpplang/parser.py

```diff
@@ -933,7 +933,7 @@ def parse_CXXNullPtrLiteralExpr(self, node) -> tree.CXXNullPtrLiteralExpr:
     @parse_debug
     def parse_NamespaceDecl(self, node) -> tree.NamespaceDecl:
         assert node['kind'] == "NamespaceDecl"
-        name = node['name']
+        name = node.get('name', "") # Namespaces can be anonymous (no name given)
```

Comment:
just `node.get("name")` if you declare `identifier? name`

Reply:
You may need to adjust `code_gen.py` though.

Reply:
Yeah I ended up using empty-string and reverted my changes to `code_gen.py` as in the end it didnt need to change.
Not sure which way is best, both should work.

Reply:
I think the `identifier? name` more cleanly converys the fact that the name is either an identifier, or no name.

Reply:
ok then, I'll go with that 👍🏽